specification/security/resource-manager/Microsoft.Security/stable/2024-08-01/defenderForStorageSettings.json

```diff
@@ -0,0 +1,285 @@
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/DefenderForStorageSettingProperties"
+        }
+      },
+      "description": "The Defender for Storage resource."
+    },
+    "DefenderForStorageSettingProperties": {
+      "type": "object",
+      "properties": {
+        "isEnabled": {
+          "type": "boolean",
+          "description": "Indicates whether Defender for Storage is enabled on this storage account."
+        },
+        "malwareScanning": {
+          "$ref": "#/definitions/MalwareScanningProperties"
+        },
+        "sensitiveDataDiscovery": {
+          "$ref": "#/definitions/SensitiveDataDiscoveryProperties"
+        },
+        "overrideSubscriptionLevelSettings": {
+          "type": "boolean",
+          "description": "Indicates whether the settings defined for this storage account should override the settings defined for the subscription."
+        }
+      },
+      "description": "Defender for Storage resource properties."
+    },
+    "MalwareScanningProperties": {
+      "type": "object",
+      "properties": {
+        "onUpload": {
+          "$ref": "#/definitions/OnUploadProperties"
+        },
+        "scanResultsEventGridTopicResourceId": {
+          "type": "string",
+          "description": "Optional. Resource id of an Event Grid Topic to send scan results to."
+        },
+        "operationStatus": {
+          "type": "object",
+          "$ref": "#/definitions/OperationStatus",
+          "readOnly": true,
+          "description": "Upon failure or partial success. Additional data describing Malware Scanning enable/disable operation."
+        }
+      },
+      "description": "Properties of Malware Scanning."
+    },
+    "OnUploadProperties": {
+      "type": "object",
+      "properties": {
+        "isEnabled": {
+          "type": "boolean",
+          "description": "Indicates whether On Upload malware scanning should be enabled."
+        },
+        "capGBPerMonth": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Defines the max GB to be scanned per Month. Set to -1 if no capping is needed."
+        },
+        "filters": {
+          "$ref": "#/definitions/OnUploadFilters",
+          "description": "Determine which blobs get scanned by On Upload malware scanning. An Or operation is performed between each filter type."
+        }
+      },
+      "description": "Properties of On Upload malware scanning."
+    },
+    "OnUploadFilters": {
+      "type": "object",
+      "properties": {
+        "excludeBlobsBeginningWith": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Exclude each blob that begins with ANY OF the given prefixes from On Upload malware scanning."
```

Comment:
>blob [](http://example.com/codeflow?start=39&length=4)

does it consider container names? In hierarchical / non-hierarchical storage accounts?